DBZ-7876 Improve NATS Jetstream sink properties for authentication (#5580)
debezium/debezium: documentation/modules/ROOT/pages/operations/debezium-server.adoc

| memory
| Controls how the messages are saved in the stream. Can be memory or file.

|[[nats-jetstream-auth-jwt]]<<nats-jetstream-auth-jwt, `debezium.sink.nats-jetstream.auth.jwt`>>
|

> |If given, enable JWT authentication with NATS. You must provide an NKey seed to use this authentication technique.

Author: Yes, both JWT authentication and password authentication works by checking the presence of their related properties.
To be more precise, you need `jwt` and `seed` to be set to enable JWT authentication.
Same behavior for `user` and `password`: those properties have to be set to enable password authentication.

|[[nats-jetstream-auth-seed]]<<nats-jetstream-auth-seed, `debezium.sink.nats-jetstream.auth.seed`>>
|
|The NKey seed to use with the JWT.

|[[nats-jetstream-auth-user]]<<nats-jetstream-auth-user, `debezium.sink.nats-jetstream.auth.user`>>
|

> |If given, enable user/password authentication with NATS. You must provide a password to use this authentication technique.

Author: Your style guidelines make sense, thank you for clarifying that out. I'm not a native english speaker and I'm still learning.

> To enable this method of authentication, is it sufficient to add the property to the configuration, or must you also specify an auth.user name?

To enable the password authentication, you need to provide both the `user` and the `password` property.

> |If given, enable user/password authentication with NATS. You must provide a password to use this authentication technique.

Reviewer: Ok, got it. So u/p authentication is enabled after you add the `user` and `password` properties to the configuration -- but with no option to provide a specific user name, I wonder should we specify the default user name through which Debezium connects to the database?

| memory
| Controls how the messages are saved in the stream. Can be memory or file.

|[[nats-jetstream-auth-jwt]]<<nats-jetstream-auth-jwt, `debezium.sink.nats-jetstream.auth.jwt`>>
|
|Add this property to the configuration to enable JSON Web Token (JWT) authentication with NATS.
To use JWT authentication with NATS, you must specify the xref:nats-jetstream-auth-seed[NKey seed].
Do not enable JWT authentication, if xref:nats-jetstream-auth-user[password authentication] is enabled.

|[[nats-jetstream-auth-seed]]<<nats-jetstream-auth-seed, `debezium.sink.nats-jetstream.auth.seed`>>
|

> |Specifies the NKey seed to use when xref:nats-jetstream-auth-jwt[JWT authentication] is enabled.

Reviewer: ```suggestion
|When xref:nats-jetstream-auth-jwt[JWT authentication] is enabled for NATS, use this property so specify the NKey seed that represents the {prodname} user.
{prodname} uses the specified NKey seed to derive a private key.
It then uses the private key to cryptographically sign the nonce challenge that the NATS server issues during the authentication process.
{prodname} returns the signed nonce to the server, along with the public key for the specified xref:nats-jetstream-auth-jwt[`debezium.sink.nats-jetstream.auth.jwt`] client.
```

| memory
| Controls how the messages are saved in the stream. Can be memory or file.

|[[nats-jetstream-auth-jwt]]<<nats-jetstream-auth-jwt, `debezium.sink.nats-jetstream.auth.jwt`>>

> |

Reviewer: ```suggestion
|No default value
```
To avoid ambiguity, when a property has no default value, explicitly set the value of the **Default** column to `No default value`.

|Specifies the identity of the NATS server client.
Add this property to the configuration to enable JSON Web Token (JWT) authentication with NATS.
To use JWT authentication with NATS, you must specify the xref:nats-jetstream-auth-seed[NKey seed].
Do not enable JWT authentication, if xref:nats-jetstream-auth-user[password authentication] is enabled.

|[[nats-jetstream-auth-seed]]<<nats-jetstream-auth-seed, `debezium.sink.nats-jetstream.auth.seed`>>

> |

Reviewer: ```suggestion
|No default value
```

|When xref:nats-jetstream-auth-jwt[JWT authentication] is enabled for NATS, use this property so specify the NKey seed that represents the {prodname} user.
{prodname} uses the specified NKey seed to derive a private key.
It then uses the private key to cryptographically sign the nonce challenge that the NATS server issues during the authentication process.
{prodname} returns the signed nonce to the server, along with the public key for the specified xref:nats-jetstream-auth-jwt[`debezium.sink.nats-jetstream.auth.jwt`] client.

|[[nats-jetstream-auth-user]]<<nats-jetstream-auth-user, `debezium.sink.nats-jetstream.auth.user`>>

> |

Reviewer: ```suggestion
|No default value
```

|Specifies the username of the authorized NAT user. +
When this property is present in the configuration, password authentication with NATS is enabled. +
To use password authentication with NATS, specify a xref:nats-jetstream-auth-password[`debezium.sink.nats-jetstream.auth.password`].
Do not enable password authentication if xref:nats-jetstream-auth-jwt[JWT authentication] is enabled.

|[[nats-jetstream-auth-password]]<<nats-jetstream-auth-password, `debezium.sink.nats-jetstream.auth.password`>>

> |

Reviewer: ```suggestion
|No default value
```
